hack/install-wrapper.sh

```diff
@@ -0,0 +1,250 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+# Constants.
+ANTREA_AGENT_BIN="antrea-agent"
+ANTREA_AGENT_CONF="antrea-agent.conf"
+INSTALL_SCRIPT="install-vm.sh"
+
+# Cloud specific constants.
+AWS="AWS"
+AZURE="AZURE"
+AWS_METADATA_QUERY_ID="http://169.254.169.254/latest/meta-data/instance-id"
+AZURE_METADATA_QUERY_ID="http://169.254.169.254/metadata/instance/compute/resourceId?api-version=2021-02-01&format=text"
+AZURE_METADATA_QUERY_NAME="http://169.254.169.254/metadata/instance/compute/name?api-version=2021-02-01&format=text"
+
+# Platform specific
+OS=""
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information"
+}
+
+_usage="Usage: $0 [arguments]
+
+Downloads install script from Antrea and setup antrea-agent on AWS or Azure VM.
+
+[arguments]
+        --ns <Namespace>                                Namespace to be used by the antrea-agent.
+        --kubeconfig <KubeconfigSavePath>               Path of the kubeconfig to access K8s API Server.
+        --antrea-kubeconfig <AntreaKubeconfigSavePath>  Path of the kubeconfig to access Antrea API Server.
+        --antrea-version <Version>                      Antrea version to be used.
+        --bin <AntreaAgentSavePath>                     Path of the antrea-agent binary(Temporary).
+        --help, -h                                      Print this message and exit."
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --ns)
+    NAMESPACE="$2"
+    shift 2
+    ;;
+    --kubeconfig)
+    KUBECONFIG="$2"
+    shift 2
+    ;;
+    --antrea-kubeconfig)
+    ANTREA_KUBECONFIG="$2"
+    shift 2
+    ;;
+    --antrea-version)
+    ANTREA_VERSION="$2"
+    shift 2
+    ;;
+    --bin)
+    AGENT_BIN="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    print_help
+    exit 1
+    ;;
+esac
+done
+
+if [ -z "$ANTREA_VERSION" ] || [ -z "$NAMESPACE" ] || [ -z "$KUBECONFIG" ] || [ -z "$ANTREA_KUBECONFIG" ]; then
+  echo "Required fields are not set"
+  print_usage
+  exit 1
+fi
+
+# Generic function to execute command with timeout.
+function execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date)"
+        echoerr "Exiting!! please check internet connectivity and re-run install script"
```

Comment:
as execute command is generic, why this log message?

Reply:
Decided to not sure execute command routine.
When we add support for multiple platforms, that I can think of making it generic and re-usable to run any commands.